plugins/test/framework.cc

```diff
@@ -155,8 +163,9 @@ proxy_wasm::WasmResult TestHttpContext::sendLocalResponse(
     uint32_t response_code, std::string_view body_text,
     proxy_wasm::Pairs additional_headers, uint32_t grpc_status,
     std::string_view details) {
-  if (phase_ != proxy_wasm::WasmHeaderMapType::RequestHeaders &&
-      phase_ != proxy_wasm::WasmHeaderMapType::ResponseHeaders) {
+  if (current_callback_ != TestHttpContext::CallbackType::RequestHeaders &&
+      current_callback_ != TestHttpContext::CallbackType::RequestBody &&
```

Comment:
This might be a good change but I'm not sure. We need to check on the Envoy behavior of immediate response during request/response body processing in STREAMED and FULL_DUPLEX body modes. Let's follow up in person.

In any case, ImmediateResponse in a body handler should be a separate PR.